lib/meilisearch/rails/multi_search.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'multi_search/result'
+
+module MeiliSearch
+  module Rails
+    class << self
+      def multi_search(searches)
+        search_parameters = searches.map do |(index_target, options)|
+          index_uid = case index_target
+                      when String, Symbol
+                        index_target
+                      else
+                        index_target.index.uid
+                      end
+
+          options.except(:class_name).merge(index_uid: index_uid)
+        end
+
+        raw_results = client.multi_search(search_parameters)['results']
+
+        MultiSearchResult.new(searches, raw_results)
+      end
```

Comment:
Just a suggestion to clean things a bit.
```suggestion
      def multi_search(searches)
        search_parameters = searches.map do |(index_target, options)|
          index_uid = find_index_target_uid(index_target)

          options.except(:class_name).merge(index_uid: index_uid)
        end

        raw_results = client.multi_search(search_parameters)['results']

        MultiSearchResult.new(searches, raw_results)
      end
      
      private
      
      def find_index_target_uid(index_target)
        case index_target
        when String, Symbol
          index_target
        else
          index_target.index.uid
        end
      end
```

Reply:
Good suggestion, I refactored it.